cfme/tests/v2v/test_rest_migrations.py

```diff
@@ -0,0 +1,120 @@
+        elif cl.name == target_cluster:
+            clusters["destination"] = "/api/clusters/{}".format(cl.id)
+    return clusters
+
+
+@pytest.fixture(scope="function")
+def get_datastores(appliance, v2v_providers):
+    datastores = {}
+    datastore = appliance.rest_api.collections.data_stores.all
+    source_ds = [
+        i.name for i in v2v_providers.vmware_provider.data.datastores if i.type == "nfs"][0]
+    target_ds = [
+        i.name for i in v2v_providers.rhv_provider.data.datastores if i.type == "nfs"][0]
+    for ds in datastore:
+        if ds.name == source_ds:
+            datastores["source"] = "/api/data_stores/{}".format(ds.id)
+        elif ds.name == target_ds:
+            datastores["destination"] = "/api/data_stores/{}".format(ds.id)
+    return datastores
+
+
+@pytest.fixture(scope="function")
+def get_networks(appliance, v2v_providers):
+    networks = {}
+    source_network = v2v_providers.vmware_provider.data.get("vlans", [None])[0]
+    target_network = v2v_providers.rhv_provider.data.get("vlans", [None])[0]
+    network = appliance.rest_api.collections.lans.all
+    for n in network:
+        if n.name == source_network:
+            networks["source"] = "/api/lans/{}".format(n.id)
+        elif n.name == target_network:
+            networks["destination"] = "/api/lans/{}".format(n.id)
+    return networks
+
+
+def test_rest_mapping_create(request, appliance, get_clusters, get_datastores, get_networks):
+    """Tests infrastructure mapping create"""
+    collection = appliance.rest_api.collections.transformation_mappings.action.create(
+        name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        state="draft",
+        transformation_mapping_items=[get_clusters, get_datastores, get_networks])[0]
+
+    @request.addfinalizer
+    def _cleanup():
+        if collection.exists:
+            collection.action.delete()
+
+    assert appliance.rest_api.response
+
+
+def test_rest_mapping_bulk_delete_from_collection(
+        request, appliance, get_clusters, get_datastores, get_networks):
+    """Tests infrastructure mapping bulk delete from collection.
+
+    Bulk delete operation deletes all specified resources that exist. When the
+    resource doesn't exist at the time of deletion, the corresponding result
+    has "success" set to false.
+    """
+    collection = appliance.rest_api.collections.transformation_mappings
+    data = [
+        {
+            "name": fauxfactory.gen_alphanumeric(),
+            "description": fauxfactory.gen_alphanumeric(),
+            "state": "draft",
+            "transformation_mapping_items": [
+                get_clusters,
+                get_datastores,
+                get_networks
+            ]
+        }
+        for _ in range(2)
+    ]
+    mapping = collection.action.create(*data)
+
+    @request.addfinalizer
+    def _cleanup():
+        for m in mapping:
```

Comment:
Question:
`L107` creating more than one items... as you are iterating for deletion?

Reply:
Yup! Here I am checking collective delete.